### ISSUE TITLE
Host Name no longer populating in tvOS 16.0 (20J373)

### ISSUE THREAD

**Issue by bondurantdt** [NU-ITS/Exhibit#14]
Hello,

Before tvOS 16 the host name set for an AppleTV would correctly populate and display. Post tvOS 16 the device name only shows "Apple TV" Confirmed this to occur on multiple devices and it persists after a full reset/wipe. Confirmed the device name is correct in the Apple TV settings.

I am going to assume this is some new privacy restriction, so I will see if there is a workaround, but if you all know of how to keep the device name correct that would be great!

**Comment by bondurantdt** [comment]
As a quick follow-up, it is a change in tvOS 16: https://getchannels.com/2022/09/15/ios16-name-issue/

**Comment by anschuetz** [comment]
Hello :) 
These unnecessary restrictions make me go mad... APPLE!!!!! 
How could this be fixed? As far as I understand the informations you linked above this would be a setting you have to do on each single device... IMHO not desirable...
To push it via appconfig-key is eventually possible if the mdm supports Variables within payloads which I do not know for Jamf.

Oh Apple... each update one more show-stopping restriction

**Comment by bondurantdt** [comment]
It looks like you can request this as an Apple entitlement for the app. I would consider this app as meeting the criteria Apple states here: https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_device-information_user-assigned-device-name

**Comment by mikegshep** [comment]
Just came across this amazing app as we are starting to deploy AppleTV's across our campus. I too updated my test AppleTV to v.16 and noticed this hiccup. Does this fix amount to the author submitting the "User Assigned Device Name Entitlement" request? Luckily the fleet of AppleTV's we purchased come with tvOS 15 out of the box but I'd feel safer if it also was cleared to work in tvOS 16.

**Comment by bondurantdt** [comment]
Yes, an entitlement is needed for this functionality to work in tvOS16. The author mentioned they are looking into this entitlement on the MacAdmins slack in the exhibit channel in mid-Sept. It is up to Apple after that...

**Comment by mikegshep** [comment]
Thanks for the update, I'll jump on over to MacAdmins slack and join the conversation.

**Comment by qharouff** [comment]
Hey! As a temporary work-around to the tvOS 16 issue, I'm working on a managed app config value to pass the name of the device. With Jamf Pro and Jamf Schools I know for sure this value will work as you're able to pass variables. I will update this issue once that has been updated but I'm hoping to test and submit and updated version of the App this week hopefully.

**Comment by rjpiper86** [comment]
Hi @qharouff, I would love a copy of what you come up with as we are JAMF school, and until Exhibit comes up with an actual fix.

Does anyone know if Exhibit is still current or has it been discontinued?

**Comment by mikegshep** [comment]
Thanks @qharouff for jumping on this issue and creating a band-aid solution for JAMF users. You rock!

**Comment by bondurantdt** [comment]
Thanks, if you need someone to Testflight my team would be happy to assist.

**Comment by qharouff** [comment]
Hi all,

I’ve submitted Exhibit version 1.1.1 to Apple. Pending Apple App Store Review, this version will have a work-around for the tvOS 16 device name issue with the new “edu.nebraska.ImageViewer.deviceName” App Config key.

In tvOS 16, Apple requires an App entitlement for accessing the user-assigned device name instead of the generic device name. Exhibit does not yet have this entitlement and therefore the name of the device will always show as “Apple TV”. This new preference value allows the setting of device name via MDM.

Jamf Pro and Jamf School both support variables in App Config so you can pass the name of the device to the Apple TV. For other platforms, you may need to check with your MDM provider to see if they support passing variables through App Config.

Jamf Pro: $DEVICENAME

```
<key>edu.nebraska.ImageViewer.deviceName</key>
<string>$DEVICENAME</string>
```

Jamf School: %Name%

```
<key>edu.nebraska.ImageViewer.deviceName</key>
<string>%Name%</string>
```
I will provide an update here when the new version has been approved and released. Updated documentation is available at: https://exhibit.readthedocs.io/

Thanks!
Quentin

**Comment by bondurantdt** [comment]
Thanks for your efforts with this!

**Comment by GrafZahl02** [comment]
Thanks a lot.
We are using exhibit with a video like the apple screensaver. It is really nice to see our school this way and provide the right informations.

**Comment by anschuetz** [comment]
Hi @qharouff 

> I’ve submitted Exhibit version 1.1.1 to Apple. Pending Apple App Store Review, (...)

I wonder if I misunderstood something or didn't get the point :)
When updating now (8 weeks later) I just have version 1.1.0
Is this due to a slow Apple or do I miss something?

best regards,
Jesko

edit: I forgot to tell you HOW MUCH I LOVE this piece of software! It is exactly, what our school missed the years before :)
Thank you very much!


2nd edit: after filling the new key, it's working with 1.1.0
thats great! THX!

**Comment by huexley** [comment]
Hi, 

Exhibit v1.1.1 I've added : 

   <key>edu.nebraska.ImageViewer.deviceName</key>
   <string>$DEVICENAME</string>

To my app config and the Apple TV name is : Mobile Device Name: ATV_C33_TEST

Sadly the app is still displaying "Apple TV", i've tried to wipe it / blank push but the name is not changing. Any suggestion ?

**Comment by qharouff** [comment]
@huexley - I’ve found out why some users are still seeing issues when using the $DEVICENAME attribute in their App Config on Jamf Pro as a work-around. When updating the device name in Jamf, the App Config capturing the device name attribute from the moment the App Config is deployed. This value isn’t automatically updating as devices are renamed. To fix this, you will need to update your App Config after renaming a device.

**Comment by mikegshep** [comment]
I've seen this same issue happen as well. In trying to make our Apple TV's super automated when they get deployed to conference rooms, I created a smart group scoped to the devices that are named 'Apple TV.' I use this for an exclusion in the Exhibit app scope so until we rename the device, it will NOT get the app. This workflow is for mainly workflows of deploying fresh AppleTVs.

**Comment by huexley** [comment]
@qharouff @mikegshep 

Thanks for you advice ! I’ve found the same solution as Mike for the newly enrolled devices. For the older ones that needs to be renamed, I remove the App, rename the ATV and put exhibit back. Hope a fix will be released (and I suspect it must comes from Jamf, not Exhibit).
